src/lib/statistics/generate_pruning_statistics.cpp

```diff
@@ -0,0 +1,94 @@
+#include "generate_pruning_statistics.hpp"
+
+#include <atomic>
+#include <iostream>
+#include <thread>
+#include <unordered_set>
+
+#include "operators/table_wrapper.hpp"
+#include "resolve_type.hpp"
+#include "statistics/column_statistics.hpp"
+#include "statistics/statistics_objects/equal_distinct_count_histogram.hpp"
+#include "statistics/statistics_objects/generic_histogram_builder.hpp"
+#include "statistics/statistics_objects/min_max_filter.hpp"
+#include "statistics/statistics_objects/null_value_ratio_statistics.hpp"
+#include "statistics/statistics_objects/range_filter.hpp"
+#include "statistics/table_statistics.hpp"
+#include "storage/create_iterable_from_segment.hpp"
+#include "storage/table.hpp"
+
+namespace {
+
+using namespace opossum;  // NOLINT
+
+template <typename T>
+void create_pruning_filter_for_segment(ColumnStatistics<T>& column_statistics, const pmr_vector<T>& dictionary) {
+  std::shared_ptr<AbstractStatisticsObject> pruning_filter;
+  if constexpr (std::is_arithmetic_v<T>) {
+    if (!column_statistics.range_filter) {
```

Comment:
I know what you mean... Those checks were there to prevent redundant work from being done, but I removed them, generate_chunk_pruning_statistics shouldn't be called multiple times anyway...